plugins/opener/src/reveal_item_in_dir.rs

```diff
@@ -0,0 +1,189 @@
+// Copyright 2019-2023 Tauri Programme within The Commons Conservancy
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-License-Identifier: MIT
+
+use std::path::Path;
+
+/// Reveal a path the system's default explorer.
+///
+/// ## Platform-specific:
+///
+/// - **Android / iOS:** Unsupported.
+pub fn reveal_item_in_dir<P: AsRef<Path>>(path: P) -> crate::Result<()> {
+    let path = path.as_ref().canonicalize()?;
+
+    #[cfg(any(
+        windows,
+        target_os = "macos",
+        target_os = "linux",
+        target_os = "dragonfly",
+        target_os = "freebsd",
+        target_os = "netbsd",
+        target_os = "openbsd"
+    ))]
+    return imp::reveal_item_in_dir(&path);
+
+    #[cfg(not(any(
+        windows,
+        target_os = "macos",
+        target_os = "linux",
+        target_os = "dragonfly",
+        target_os = "freebsd",
+        target_os = "netbsd",
+        target_os = "openbsd"
+    )))]
+    Err(crate::Error::UnsupportedPlatform)
+}
+
+#[cfg(windows)]
+mod imp {
+    use super::*;
+
+    use windows::{
+        core::{w, HSTRING, PCWSTR},
+        Win32::{
+            Foundation::ERROR_FILE_NOT_FOUND,
+            System::Com::CoInitialize,
+            UI::{
+                Shell::{
+                    ILCreateFromPathW, ILFree, SHOpenFolderAndSelectItems, ShellExecuteExW,
+                    SHELLEXECUTEINFOW,
+                },
+                WindowsAndMessaging::SW_SHOWNORMAL,
+            },
+        },
+    };
+
+    pub fn reveal_item_in_dir(path: &Path) -> crate::Result<()> {
+        let file = dunce::simplified(path);
+
+        let _ = unsafe { CoInitialize(None) };
+
+        let dir = file
+            .parent()
+            .ok_or_else(|| crate::Error::NoParent(file.to_path_buf()))?;
+
+        let dir = HSTRING::from(dir);
+        let dir_item = unsafe { ILCreateFromPathW(PCWSTR::from_raw(dir.as_ptr())) };
+
+        let file_h = HSTRING::from(file);
+        let file_item = unsafe { ILCreateFromPathW(PCWSTR::from_raw(file_h.as_ptr())) };
+
+        unsafe {
+            if let Err(e) = SHOpenFolderAndSelectItems(dir_item, Some(&[file_item]), 0) {
+                if e.code().0 == ERROR_FILE_NOT_FOUND.0 as i32 {
```

Comment:
We can simplify this

```suggestion
        let dir_item = unsafe { ILCreateFromPathW(&HSTRING::from(dir)) };

        let file_h = HSTRING::from(file);
        let file_item = unsafe { ILCreateFromPathW(&file_h) };

        unsafe {
            if let Err(e) = SHOpenFolderAndSelectItems(dir_item, Some(&[file_item]), 0) {
                if e == ERROR_FILE_NOT_FOUND.into() {
```

Also it there any reason to fallback to `ShellExecute` on `ERROR_FILE_NOT_FOUND`? If we do need to fallback to it, we can't do `?` on `ShellExecuteExW` to early return here, or we'll leak `dir_item` and `file_item`

Reply:
fixed the leak and simplified the code